esp/src/eclwatch/QuerySetQueryWidget.js

```diff
@@ -475,18 +498,64 @@ define([
             registry.byId(this.id + "Activate").set("disabled", !isActive);
             registry.byId(this.id + "Deactivate").set("disabled", !isNotActive);
             registry.byId(this.id + "Open").set("disabled", !hasSelection);
+            registry.byId(this.id + "RecreateQueryDropDown").set("disabled", !hasSelection);
+
 
 
             this.menuUnsuspend.set("disabled", !isNotSuspended);
             this.menuSuspend.set("disabled", !isSuspended);
             this.menuActivate.set("disabled", !isActive);
             this.menuDeactivate.set("disabled", !isNotActive);
+
+            if (hasSelection) {
+                arrayUtil.forEach(selection, function (item, idx) {
+                    data.push(item);
+                });
+                this.recreateQueriesGrid.setData(data);
+            }
          },
 
         _onRefresh: function (params) {
            this.refreshGrid();
         },
 
+        _onRecreateQueriesSuccess: function (status) {
+            var context = this;
+            if (status) {
+                dojo.publish("hpcc/brToaster", {
+                    Severity: "Message",
+                    Source: "WsWorkunits.WURecreateQuery",
+                    Exceptions: [{ Source: context.i18n.RecreateQuery, Message: context.i18n.SuccessfullySaved }]
+                });
+            }
+        },
+
+        _onRecreateQueries: function () {
+            if (this.recreateForm.validate()) {
+                var context = this;
+                var success = false;
+                arrayUtil.forEach(this.recreateQueriesGrid.store.data, function (item, idx) {
+                        var request = domForm.toObject(context.id + "RecreateForm");
+                        request.Republish === "on" ? request.Republish = 0 : request.Republish = 1;
+                        request.AllowForeignFiles = "on" ? request.AllowForeignFiles = 0 : request.AllowForeignFiles = 1;
+                        request.UpdateDfs = "on" ? request.UpdateDfs = 0 : request.UpdateDfs = 1;
```

Comment:
Should some of these be "==" or "===", rather than "="?

Reply:
Yes, the first one comparison in the ternary should be I will update.